app/views/onboarding/Onboarding5.js

```diff
@@ -1,72 +1,403 @@
+  }
+
+  /**
+   * Helper method to determine the next step for permission requests.
+   * In general there is a linear flow, but because Android does not
+   * require permission for notifications, we skip the notifications
+   * step on Android.
+   *
+   * @returns StepEnum
+   */
+  getNextStep() {
+    switch (this.state.currentStep) {
+      case StepEnum.LOCATION:
+        return isPlatformiOS()
+          ? StepEnum.NOTIFICATIONS
+          : StepEnum.HCA_SUBSCRIPTION;
+      case StepEnum.NOTIFICATIONS:
+        return StepEnum.HCA_SUBSCRIPTION;
+      case StepEnum.HCA_SUBSCRIPTION:
+        return StepEnum.DONE;
+    }
+  }
+
+  checkLocationStatus() {
+    // NEED TO TEST ON ANDROID
+    const nextStep = this.getNextStep(StepEnum.LOCATION);
+    const locationPermission = this.getLocationPermissionSetting();
+
+    check(locationPermission)
+      .then(result => {
+        switch (result) {
+          case RESULTS.GRANTED:
+            this.setState({
+              currentStep: nextStep,
+              locationPermission: PermissionStatusEnum.GRANTED,
+            });
+            break;
+          case RESULTS.UNAVAILABLE:
+          case RESULTS.BLOCKED:
+            this.setState({
+              currentStep: nextStep,
+              locationPermission: PermissionStatusEnum.DENIED,
+            });
+            break;
+        }
+      })
+      .catch(error => {
+        console.log('error checking location: ' + error);
+      });
+  }
+
+  checkNotificationStatus() {
+    const nextStep = this.getNextStep(StepEnum.NOTIFICATIONS);
+
+    checkNotifications().then(({ status }) => {
+      switch (status) {
+        case RESULTS.GRANTED:
+          this.setState({
+            currentStep: nextStep,
+            notificationPermission: PermissionStatusEnum.GRANTED,
+          });
+          break;
+        case RESULTS.UNAVAILABLE:
+        case RESULTS.BLOCKED:
+          this.setState({
+            currentStep: nextStep,
+            notificationPermission: PermissionStatusEnum.DENIED,
+          });
+          break;
       }
     });
-  };
+  }
+
+  async checkSubsriptionStatus() {
+    const hasUserSetSubscription = await HCAService.hasUserSetSubscription();
+    const isEnabled = await HCAService.isAutosubscriptionEnabled();
+    const nextStep = this.getNextStep(StepEnum.HCA_SUBSCRIPTION);
 
-  onGoToSettingsPress = () => {
-    this.props.navigation.navigate('ChooseProviderScreen');
-  };
+    // Only update state if the user has already set their subscription status
+    if (hasUserSetSubscription) {
+      const permission = isEnabled
+        ? PermissionStatusEnum.GRANTED
+        : PermissionStatusEnum.DENIED;
 
-  onButtonPress = () => {
-    this.props.navigation.replace('Onboarding6');
-  };
+      this.setState({
+        currentStep: nextStep,
+        authSubscriptionStatus: permission,
+      });
+    }
+  }
+
+  getLocationPermissionSetting() {
```

Comment:
this feels like it should be in a service. the view doesn't care if iOS has a different permission string.

Reply:
Cool if we hold off on this for the larger body of work you mentioned in Slack?